core/node/base_token_fetcher/src/lib.rs

```diff
@@ -119,11 +128,21 @@ impl BaseTokenFetcher {
 
 #[async_trait]
 impl ConversionRateFetcher for BaseTokenFetcher {
-    fn conversion_rate(&self) -> anyhow::Result<u64> {
-        anyhow::Ok(
-            self.latest_to_eth_conversion_rate
-                .load(std::sync::atomic::Ordering::Relaxed),
-        )
+    fn conversion_rate(&self) -> anyhow::Result<BigDecimal> {
+        let lock = match self.latest_to_eth_conversion_rate.lock() {
```

Comment:
`std` mutex only returns an error if the mutex has been poisoned somewhere else, e.g. other thread has panicked while holding the mutex guard. It is an extreme situation that is extremely hard to handle gracefully. It is safe to simply unwrap the lock here.

Reply:
Removed the `match` in favour of `unwrap`ing the lock. The tokio `mutex`es that are now std also `unwrap` now.